File: solutions/java/squeaky-clean/1/src/main/java/SqueakyClean.java
class SqueakyClean {
    static String clean(String identifier) {
        char[] characters = identifier.toCharArray();
        StringBuilder sb = new StringBuilder();

        boolean upCaseNextChar = false;
        
        for (int i = 0; i < characters.length; i++) {
            char character = characters[i];
            
            if (character == ' ') {
                sb.append('_');
            } else if (character == '-') {
                upCaseNextChar = true;
            } else if(character == '4') {
                sb.append('a');
            } else if(character == '3') {
                sb.append('e');
            } else if(character == '0') {
                sb.append('o');
            } else if(character == '1') {
                sb.append('l');
            } else if(character == '7') {
                sb.append('t');
            } else if (Character.isLetter(character)){
                if (upCaseNextChar) {
                    sb.append(Character.toUpperCase(character));
                    upCaseNextChar = false;
                } else {
                    sb.append(character);                
                }
            }
        }

        return sb.toString();
    }
}

File: solutions/csharp/squeaky-clean/1/SqueakyClean.cs
using System.Text;

public static class Identifier
{
    private static bool IsGreekLowerCase(char c) => (c >= 'α' && c <= 'ω');

    public static string Clean(string identifier)
    {
        var sb = new StringBuilder();

        var isAfterDash = false;

        foreach (var c in identifier)
        {
            sb.Append(c switch
            {
                _ when IsGreekLowerCase(c) => default,
                _ when isAfterDash => char.ToUpperInvariant(c),
                _ when char.IsLetter(c) => c,
                _ when char.IsWhiteSpace(c) => '_',
                _ when char.IsControl(c) => "CTRL",
                _ => default
            });
            
            isAfterDash = c.Equals('-');
        }

        return sb.ToString();
    }
}
